File: e2e-tests/src/test/java/e2e/testUtil/KafkaProps.java
package e2e.testUtil;


import org.apache.kafka.clients.producer.ProducerConfig;

import java.util.HashMap;
import java.util.Map;
import java.util.Properties;

public class KafkaProps {
    private final String bootstrapServers;
    private final Map<String, Object> properties = new HashMap<>();

    public KafkaProps(String bootstrapServers) {
        this.bootstrapServers = bootstrapServers;
    }

    public void put(String key, Object value) {
        properties.put(key, value);
    }

    public Properties toProperties() {
        Properties props = new Properties();
        props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
        props.putAll(properties);
        return props;
    }
}

File: e2e-tests/src/main/kotlin/testUtil/KafkaProps.kt
package testUtil

import org.apache.kafka.clients.producer.ProducerConfig
import java.util.Properties

class KafkaProps(private val bootstrapServers: String) {
    private val properties = mutableMapOf<String, Any>()

    fun put(key: String, value: Any) {
        properties[key] = value
    }

    fun toProperties(): Properties = Properties().apply {
        put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers)
        putAll(properties)
    }
}

